catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -110,14 +118,43 @@ public SourceResponse query(QueryRequest request) throws UnsupportedQueryExcepti
 
         SolrQuery query = getSolrQuery(request, filterDelegateFactory.newInstance(resolver));
 
+        boolean isFacetedQuery = false;
+        Serializable textFacetPropRaw = request.getPropertyValue(EXPERIMENTAL_FACET_PROPERTIES_KEY);
+
+        if (textFacetPropRaw != null && textFacetPropRaw instanceof FacetProperties) {
+            FacetProperties textFacetProp = (FacetProperties) textFacetPropRaw;
+            isFacetedQuery = true;
+
+            textFacetProp.getFacetAttributes()
```

Comment:
:question: Do we have any guarantees that `textFacetProp` is not null before being used here?